src/connect/printer.cpp

```diff
@@ -7,16 +7,73 @@
 using std::make_tuple;
 using std::tuple;
 
+namespace {
+
+struct Crc {
+    uint32_t crc = 0;
+
+    template <class T>
+    Crc &add(const T &value) {
+        crc = crc32_calc_ex(crc, reinterpret_cast<const uint8_t *>(&value), sizeof value);
+        return *this;
+    }
+
+    Crc &add_str(const char *s) {
+        crc = crc32_calc_ex(crc, reinterpret_cast<const uint8_t *>(s), strlen(s));
+        return *this;
+    }
+
+    uint32_t done() const {
+        return crc;
+    }
+};
+
+}
+
 namespace connect_client {
 
+uint32_t Printer::Params::telemetry_fingerprint(bool include_xy_axes) const {
+    // Note: keep in sync with the rendering of telemetry in render.cpp
+    // Note: There are some guessed "precision" constants - making sure we
+    //   don't resend the telemetry too often because something changes only a
+    //   little bit.
+
+    Crc crc;
+
+    if (include_xy_axes) {
+        // Add the axes, but with only whole-int precision.
+        crc
+            .add(int(pos[Printer::X_AXIS_POS]))
+            .add(int(pos[Printer::Y_AXIS_POS]));
+    }
+
+    return crc
+        .add(int(pos[Printer::Z_AXIS_POS]))
+        .add(print_speed)
+        .add(flow_factor)
+        .add(job_id)
+        // The RPM values are in thousands and fluctuating a bit, we don't want
+        // that to trigger the send too often, only when it actually really
+        // changes.
```

Comment:
I don't see any mechanism by which you would achieve this, am I missing something, or did you forget to divide the rpm by some amount as you do with the `filament_used`?

Reply:
That's when I'm testing with a device that has no fans connected :facepalm: Of course.